x-pack/dockerlogbeat/pipelinemanager/clientLogReader_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemanager
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/docker/docker/daemon/logger"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/x-pack/dockerlogbeat/pipelinemock"
+	"github.com/elastic/beats/x-pack/dockerlogbeat/pipereader"
+)
+
+func TestNewClient(t *testing.T) {
+	logString := "This is a log line"
+	client, teardown := setupTestClient(t, logString)
+	defer teardown()
+
+	event := testReturn(t, client)
+	assert.Equal(t, event.Fields["message"], logString)
+}
+
+func setupTestClient(t *testing.T, logString string) (*pipelinemock.MockPipelineConnector, func()) {
```

Comment:
nit; the function name does not match the return type. Where do we create and publish the event?

Reply:
Alright, cleaned this up and adding some comments that hopefully clarify this.